### ISSUE TITLE
for-in assignment issue

### ISSUE THREAD

**Issue by nowindowsowrking** [jquery/esprima#1796]
<!--
### Steps to reproduce

```js
esprima.parse('for (x in [], []) {}')
```

### Expected output
Should throw an error

### Actual output
Parses just fine

### Solution
Change [this line](https://github.com/jquery/esprima/blob/master/src/parser.ts#L2321) into this
` right = this.parseAssignmentExpression();`

**Comment by michaelficarra** [comment]
That's not correct. For-in allows any expression on the right. For-of only allows assignment expressions and higher on the right.

**Comment by nowindowsowrking** [comment]
@michaelficarra Agree! I overlooked a few other issues in the code that caused my first solution to "solve" it. I let the author handle this.

**Comment by nowindowsowrking** [comment]
@mikesherov It's actual still an issue and not solved. It should not parse.

**Comment by mikesherov** [comment]
Even with what @michaelficarra said?

**Comment by nowindowsowrking** [comment]
@mikesherov I double checked it with my own parser. Correct. Not an bug. It should parse. So ignore this. And for future I will validate against my own parser before posting issues.
